pkg/controller/clusterdeprovision/clusterdeprovision_controller.go

```diff
@@ -135,6 +137,25 @@ func (r *ReconcileClusterDeprovision) Reconcile(request reconcile.Request) (reco
 		return reconcile.Result{}, nil
 	}
 
+	// Check if there is a ClusterDeployment owning this Deprovision, if so look it up and
+	// make sure it has a deletion timestamp. Otherwise bail out as a safety check.
```

Comment:
The controller should also bail out if the deprovision is **not owned** by a ClusterDeployment.

Reply:
I wasn't sure where to go with that as technically at one point this was designed and used to be able to cleanup clusters that were provisioned with hive and the preserveOnDelete option, then later deprovision them without there being a cluster deployment. Feature was axed in OCM but might make a reappearance so it was theoretically possible that there could be a detached unassociated Deprovision.

Reply:
I'm not keen on being in the business of providing a means to delete a random cluster not controller by Hive. I would rather prevent the deletion now and relax the restriction in the future if we decide to explicitly allow the feature.

Reply:
Deal.

Reply:
Not thinking about it too much, but I think I would feel better if, to cover the use case described, the user had to create a new clusterdeployment that adopted the existing cluster.